src/WSRequestHandler_Transitions.cpp

```diff
@@ -120,3 +120,53 @@ RpcResponse WSRequestHandler::GetTransitionDuration(const RpcRequest& request) {
 	obs_data_set_int(response, "transition-duration", obs_frontend_get_transition_duration());
 	return request.success(response);
 }
+
+/**
+ * Set the manual position of the T-Bar (in Studio Mode) to the specified value. Will return an error if OBS is not in studio mode
+ * or if the current transition doesn't support T-Bar control.
+ *
+ * @param {double} `position` T-Bar position. This value must be between 0.0 and 1.0.
+ *
+ * @api requests
+ * @name SetTBarPosition
+ * @category transitions
+ * @since 4.8.0
+ */
+RpcResponse WSRequestHandler::SetTBarPosition(const RpcRequest& request) {
+	if (!obs_frontend_preview_program_mode_active()) {
+		return request.failed("studio mode not enabled");
+	}
+
+	OBSSourceAutoRelease currentTransition = obs_frontend_get_current_transition();
+	if (obs_transition_fixed(currentTransition)) {
+		return request.failed("current transition doesn't support t-bar control");
+	}
+
+	double position = obs_data_get_double(request.parameters(), "position");
+	if (position < 0.0 || position > 1.0) {
+		return request.failed("position is out of range");
+	}
+
+	obs_transition_set_manual_time(currentTransition, position);
```

Comment:
This is a big problem, and the `SetTBarPosition` feature will not be worth adding unless it is fixed.

The problem is that in the OBS UI, when you change the t-bar, it does a whole bunch of important stuff before calling this. See [here](https://github.com/cg2121/obs-studio/blob/68d88fa4c6e28edf2368a38ad8bc1f2403d0231d/UI/window-basic-main-transitions.cpp#L921)

So what actually has to happen is to first call `TBarChanged`, then `TBarReleased` needs to be called to clean up the changes.